File: uniquePaths.py
class Solution:
    # @return an integer
    def uniquePaths(self, m, n):
        t = [1 for i in range(n)]
        k = [1 for j in range(n)]
        for i in range(m-1):
        	for j in range(n-2, -1, -1):
        		k[j] = k[j+1] + t[j]
        	t = k
        	k = [1 for j in range(n)]
        return t[0]

s = Solution()
print s.uniquePaths(2,2)

File: JAVA/uniquePaths.java
public class Solution {
    public int uniquePaths(int m, int n) {
        if (n == 1||m == 1)
            return 1;
        int[] lastLine = new int[n];
        int[] currentLine = new int[n];
        
        for (int k = 0; k < n; k++)
            currentLine[k] = 1;
            
        int count = 1;
        while (count < m){
            for (int j = 1; j < n; j++)
                lastLine[j] = currentLine[j];
            for (int i = 1; i < n; i++)
                currentLine[i] = currentLine[i-1] + lastLine[i];
            count++;
        }
        return currentLine[n-1];
    }
}